tests/testthat/test-zzz.R

```diff
@@ -18,6 +18,36 @@ test_that("$ gives useful error", {
   expect_equal(x$x, 1)
 })
 
+test_that("[ gives more accurate error", {
+  foo <- new_class("foo")
+  x <- foo()
+  expect_snapshot(error = TRUE, {
+    x[1]
+    x[1] <- 1
+  })
```

Comment:
For this sort of very succinct example, I'd be tempted to do `x <- new_class("foo")()` inside the snapshot, just so everything is self-contained.